python/cudf/cudf/bindings/join.pyx

```diff
@@ -229,11 +200,10 @@ cpdef join(col_lhs, col_rhs, left_on, right_on, how, method):
                 valids.append(None)
```

Comment:
@rgsl888prabhu do you mind cleaning up this section of the code while we're changing things? We have helper functions to convert gdf_column or cudf::table objects into cudf Python objects that should be used here.